src/graphql/schema/types/pagination.ts

```diff
@@ -0,0 +1,117 @@
+import { Max, Min } from 'class-validator'
+import * as Relay from 'graphql-relay'
+
+import { ArgsType, ClassType, Field, Int, ObjectType } from 'type-graphql'
+import { dbClient } from 'data/config'
+
+@ArgsType()
+export class OffsetPaginationArgs {
+  @Field(() => Int, { nullable: true })
+  @Min(1)
+  @Max(50)
+    take?: number
+
+  @Field(() => Int, { nullable: true })
+  @Min(1)
+  @Max(50)
+    skip?: number
+}
+
+@ArgsType()
+export class ConnectionArgs implements Relay.ConnectionArguments {
```

Comment:
Here we're using the connection types pattern, originally introduced by Relay

Connection types expose two fields:
- `edges`: Contains requested data, but does not return the items immediately and instead returns an edge
type with extra connection metadata for that particular item
- `pageInfo`: Contains metadata about the pagination itself